### DISCUSSION TITLE
Forcing Expo SDK to use FCM HTTP v1 API

### DISCUSSION THREAD

**Discussion by hramchandani** [ctwillie/expo-server-sdk-php#41] [Q&A] [answered]
Good day.

I am updating my application to use FCM HTTP v1 API as requested by the Expo team. See this post: https://expo.dev/blog/expo-adds-support-for-fcm-http-v1-api

Towards the end of the article the Expo team states we can test out our use of the v1 API by adding ?useFcmV1=true in the request URL when sending the push notification.

My question is how can we test out the use of the v1 API via the expo-server-sdk-php. Does the sdk already use the v1 API? Is there a parameter we must pass in to force it to use the v1 API?

Any information you can provide would be appreciated. Thank you in advance.

**Comment by ctwillie** [comment, marked as answer]
Hey @hramchandani !!

I'm unable to read the blog you linked to. An error is preventing that page from loading at the moment.

 It seems I didn't make the `Expo` class as extensible as it could be when I wrote this package.. But try the below to see if it works and/or meets your needs.

```php
use ExpoSDK\Expo;
use ExpoSDK\DriverManager;

$driverManager = new DriverManager('file');

$httpClientOptions = [
  'query' => [
      'useFcmV1' => true,
    ],
];

$expo = new Expo($driverManager, $httpClientOptions);
$expo->send(...)->to(...)->push();
```

 Again, I'm not %100 sure this will work given the implementation I've employed.. but I would start here.

**Comment by ctwillie** [comment]
Adding `on_stats` to Guzzle will help confirm the requested url is what you are looking for...


```php
use ExpoSDK\Expo;
use ExpoSDK\DriverManager;
use GuzzleHttp\TransferStats;

$driverManager = new DriverManager('file');

$httpClientOptions = [
  'query' => [
      'useFcmV1' => true,
    ],
  'on_stats' => function (TransferStats $stats) {
        echo $stats->getEffectiveUri();
    }
];

$expo = new Expo($driverManager, $httpClientOptions);
$expo->send(...)->to(...)->push();
```

**Reply by mgscreativa** [reply]
This works just fine!
